R/extract.R

```diff
@@ -128,13 +142,23 @@ extract_preprocessor.workflow_set <- function(x, id, ...) {
 #' @rdname extract_workflow_set_result
 extract_parameter_set_dials.workflow_set <- function(x, id, ...) {
    y <- filter_id(x, id)
+
+   if ("param_info" %in% names(y$option[[1]])) {
+      return(y$option[[1]][["param_info"]])
+   }
+
    extract_parameter_set_dials(y$info[[1]]$workflow[[1]])
 }
 
 #' @export
 #' @rdname extract_workflow_set_result
 extract_parameter_dials.workflow_set <- function(x, id, parameter, ...) {
    y <- filter_id(x, id)
+
+   if ("param_info" %in% names(y$option[[1]])) {
```

Comment:
Just a suggestion: what do you think of using `extract_parameter_set_dials.workflow_set()` in here to extract the right set and then just extract the parameter from the set. that way we'd only have to write the "look at the options" logic once in `extract_parameter_set_dials.workflow_set()`. It wouldn't be as obvious what going on when reading only that function but the other one is right above it and we'd need to only maintain that logic in one place.

Reply:
I think that's a great idea.